src/libraries/Common/src/System/Security/Cryptography/AesGcm.cs

```diff
@@ -0,0 +1,353 @@
+    /// </summary>
+    public sealed partial class AesGcm : IDisposable
+    {
+        private const int NonceSize = 12;
+
+        /// <summary>
+        /// Gets the nonce sizes, in bytes, supported by this instance.
+        /// </summary>
+        /// <value>
+        /// The nonce sizes supported by this instance: 12 bytes (96 bits).
+        /// </value>
+        public static KeySizes NonceByteSizes { get; } = new KeySizes(NonceSize, NonceSize, 1);
+
+        /// <summary>
+        /// Gets the size of the tag, in bytes.
+        /// </summary>
+        /// <value>
+        /// The size of the tag that must be used for encryption or decryption, or <see langword="null" /> if the
+        /// tag size is unspecified.
+        /// </value>
+        public int? TagSizeInBytes { get; }
+
+        /// <summary>
+        /// Gets the tag sizes, in bytes, supported by this instance.
+        /// </summary>
+        /// <value>
+        /// The tag sizes supported by this instance: 12, 13, 14, 15, or 16 bytes (96, 104, 112, 120, or 128 bits).
+        /// </value>
+        public static partial KeySizes TagByteSizes { get; }
+
+        /// <summary>
+        /// Gets a value that indicates whether the algorithm is supported on the current platform.
+        /// </summary>
+        /// <value>
+        /// <see langword="true"/> if the algorithm is supported; otherwise, <see langword="false"/>.
+        /// </value>
+        public static partial bool IsSupported { get; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="AesGcm" /> class with a provided key and required tag size.
+        /// </summary>
+        /// <param name="key">The secret key to use for this instance.</param>
+        /// <param name="tagSizeInBytes">The size of the tag, in bytes, that encryption and decryption must use.</param>
+        /// <exception cref="CryptographicException">
+        ///   The <paramref name="key" /> parameter length is other than 16, 24, or 32 bytes (128, 192, or 256 bits).
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   The <paramref name="tagSizeInBytes" /> parameter is an unsupported tag size indicated by
+        ///   <see cref="TagByteSizes" />.
+        /// </exception>
+        /// <exception cref="PlatformNotSupportedException">
+        ///   The current platform does not support AES-GCM.
+        /// </exception>
+        /// <remarks>
+        ///   The <paramref name="tagSizeInBytes" /> parameter is used to indicate that the tag parameter in <c>Encrypt</c>
+        ///   or <c>Decrypt</c> must be exactly this size. Indicating the required tag size prevents issues where callers
+        ///   of <c>Decrypt</c> may supply a tag as input and that input is truncated to an unexpected size.
+        /// </remarks>
+        public AesGcm(ReadOnlySpan<byte> key, int tagSizeInBytes)
```

Comment:
This is the AesGcm implementation in `Common`, so it lacks the constructors without `tagSizeInBytes`. Those constructors are only in the S.S.Cryptography implementation.

Here is the resulting public API surface for .NET Standard 2.0 and .NET Framework:
 
<img width="664" alt="Screenshot 2025-01-22 at 3 46 13 PM" src="https://github.com/user-attachments/assets/5599b5d1-f610-4309-8fda-c4e84b09deb9" />